### DISCUSSION TITLE
Does AudioHandler get triggered in OffScreen browser?

### DISCUSSION THREAD

**Discussion by yaskovdev** [cefsharp/CefSharp#4028] [Q&A] [answered]
Good evening!

I am trying to capture audio from [the Web page](https://github.com/yaskovdev/sandbox/blob/master/headful-browser-audio-video-capturing-server/index.html#L19) rendered in an instance of a _headless_ Chromium browser (using CefSharp.OffScreen to run it). The Web page plays audio automatically once opened. The browser is configured [to allow it](https://github.com/yaskovdev/sandbox/blob/master/HeadlessBrowserAudioVideoCapturing/HeadlessBrowserAudioVideoCapturing/Program.cs#L12).

I am using an instance of `IAudioHandler` [in this way](https://github.com/yaskovdev/sandbox/blob/master/HeadlessBrowserAudioVideoCapturing/HeadlessBrowserAudioVideoCapturing/Program.cs#L15) (just like in the examples, basically). However, none of the methods of the audio handler get called.

The absence of the `IAudioHandler` example for [CefSharp.OffScreen](https://github.com/cefsharp/CefSharp/tree/master/CefSharp.OffScreen.Example) suggests that headless browser does not support the handler.

Does CefSharp.OffScreen support audio handlers? If no, what is the reason that it does not?

Thank you!

**Comment by amaitland** [comment, marked as answer]
Tye OffScreen implementation starts as being muted by default. Call https://github.com/cefsharp/CefSharp/blob/cefsharp/98/CefSharp.OffScreen/CefSettings.cs#L29 to enable audio.

**Reply by yaskovdev** [reply]
Wow, I [enabled audio](https://github.com/yaskovdev/sandbox/blob/master/HeadlessBrowserAudioVideoCapturing/HeadlessBrowserAudioVideoCapturing/Program.cs#L13) as you suggested and the handler started to be called.

Thank you!
